server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java

```diff
@@ -335,19 +306,27 @@ public static void decodeOneRowToPreparedStatement(OneRow row, PreparedStatement
         }
     }
 
-    private static final Logger LOG = LoggerFactory.getLogger(JdbcResolver.class);
+    private static final Set<DataType> DATATYPES_SUPPORTED = EnumSet.of(
+            DataType.VARCHAR,
+            DataType.BPCHAR,
+            DataType.TEXT,
+            DataType.BYTEA,
+            DataType.BOOLEAN,
+            DataType.INTEGER,
+            DataType.FLOAT8,
+            DataType.REAL,
+            DataType.BIGINT,
+            DataType.SMALLINT,
+            DataType.NUMERIC,
+            DataType.TIMESTAMP,
+            DataType.DATE
+    );
 
-    // SimpleDateFormat to parse TEXT into DATE
-    private static ThreadLocal<SimpleDateFormat> dateSDF = ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy-MM-dd"));
-    // SimpleDateFormat to parse TEXT into TIMESTAMP (with microseconds)
-    private static ThreadLocal<SimpleDateFormat[]> timestampSDFs = ThreadLocal.withInitial(() -> new SimpleDateFormat[]{
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSS"),
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSS"),
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSS"),
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS"),
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SS"),
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S"),
-            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss"),
-            new SimpleDateFormat("yyyy-MM-dd")
-    });
+    // SimpleDateFormat objects to parse TEXT into DATE
+    private final SimpleDateFormat dateSDF = new SimpleDateFormat("yyyy-MM-dd");

```

Comment:
You can use the following instead, not sure if it will impact performance though. Maybe try it?
```suggestion
    private static final String DATE_FORMATTER_BASE_PATTERN = "yyyy-MM-dd HH:mm:ss";
    static final DateTimeFormatter DATE_FORMATTER =
            new DateTimeFormatterBuilder().appendPattern(DATE_FORMATTER_BASE_PATTERN)
                    // Parsing nanos in strict mode, the number of parsed digits must be between 0 and 6 (millisecond support)
                    .appendFraction(ChronoField.NANO_OF_SECOND, 0, 6, true).toFormatter();
```

Reply:
Thank you. I tested the perfomance of the fix; no changes were noted.

Reply:
@leskin-in -- great performance improvement ! Coming down from 30s to 5s is a factor of 6 !! What was the bottleneck -- threadlocal or the old formatter or both ?

Reply:
Both. The old parsing logic could also throw up to six exceptions (and it did in case of our dataset) that should have been processed.